keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/constraints/PipelineConstraintEvaluator.kt

```diff
@@ -67,6 +69,23 @@ class PipelineConstraintEvaluator(
 
     val judge = "pipeline:${constraint.pipelineId}"
 
+    val currentVersion = keelRepository
+      .getCurrentVersionInEnv(deliveryConfig, targetEnvironment.name, artifact.reference)
+
+    if (currentVersion != version) {
+      repository
```

Comment:
I'd add a log.warn statement here, since we don't typically expect this to happen.